pkg/datarepair/queue/queue_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage/redis"
+	"storj.io/storj/storage/redis/redisserver"
+	"storj.io/storj/storage/teststore"
+)
+
+func newTestQueue(t *testing.T) (*Queue, func()) {
+	addr, cleanup, err := redisserver.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, err := redis.NewClient(addr, "", 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	queue := NewQueue(client)
+	return queue, cleanup
+}
+
+func TestEnqueueDequeue(t *testing.T) {
+	queue, cleanup := newTestQueue(t)
+	defer cleanup()
+
+	seg := &pb.InjuredSegment{
+		Path:       "abc",
+		LostPieces: []int32{},
+	}
+	err := queue.Enqueue(seg)
```

Comment:
Add 100 items with `Path: "a" + strconv.Itoa(i)`. That way you can verify two things: 1. that enqueue/dequeue are working and 2. that they are returning items in the correct order.

Reply:
why do you suggest prepending an "a" to the path string?

Reply:
Ah nevermind, doesn't matter here. I think the S3 compatibility layer had a restriction that it cannot have it start with a number... although that might have been only for buckets.